HW06.Rmd

```diff
@@ -61,28 +83,74 @@ detach(package:EnsDb.Hsapiens.v75, unload = T)
 
 GBM.genes.of.interest <- filter(ens.mapped_GBM, gene_symbol %in% genes_of_interest)
 
 #Now perform a t-test between the H3K mutated and the wt samples. There are many ways to do this actually, you can use a for loop or you could do the tidy alternative with broom(), but the for loop is probably the easiest
 
-#print out the t-test results
-```
+# I initially tried the following codes but I kept encountering an error with the filter function. 
+
+#for (gene in genes_of_interest) {
+#  WT <- GBM.genes.of.interest %>%
+#  filter(gene_symbol == gene, type == "WT")
+  
+#  mutated <- GBM.genes.of.interest %>%
+#  filter(gene_symbol == gene, type == "H3K27M_mutated")
+#}
 
+# I decided to go a different route and reorganize the table first this time. 
 
+reorg.GBM.genes.of.interest <- GBM.genes.of.interest %>% 
+  gather(key = "key", value = "value", -gene_symbol)  %>%
+  separate(key, into = c ("group", "order"), convert = TRUE)
+
+# Now that I have the table reorganized, I created a for loop to fo through each gene and run a ttest between mutated and WT.
+
+for (gene in genes_of_interest) {
+  stat_df <- reorg.GBM.genes.of.interest  %>%
+    filter(gene_symbol == gene)
+  
+#print out the t-test results
+  print(paste("........", gene, "........"))
+  print(t.test(value ~ group, stat_df))
+}
+```
 
 Now create a graphing function to create boxplots to visualize the results. Plot expression on the y-axis. The graph should look like [this example](IRX1_example.png)
 
 ```{r create a graphing function}
 #to work in the tidyverse, it will be easier to make tidy the dataframe first
+#create a graphing function
+graphhh <- function(gene) {
+  stat_df <- GBM.genes.of.interest %>%
+    filter(gene_symbol == gene)
+  
+    boxplot <- ggplot(stat_df, aes(x = group, y = value, fill = group)) +
+      geom_boxplot() +
+      labs(x = "H3K27M", y = "Expression_(TPM)",
+        title = paste0(gene, " Expression in GBM models\nby H3K27M Mutated or WT Status")) +
+      theme(legend.position = "none", plot.title = element_text(hjust = 0.5)) +
+      theme_bw() 
+    
+    print(boxplot)
+}
 
+#then use a for loop combined with the graphing function to make a graph for all your genes of interest 
+for (gene in genes_of_interest) {
+  graphhh(gene)
+}
 
+# I am almost there but I keep getting an error message that says:
+#Error in FUN(X[[i]], ...) : object 'group' not found
 
-#create a graphing function
+# Upon further inspection, I realized something wrong in my global environment after running my first for loop in my third chunk of code (for gene in genes_of_interest). I was able to run the t-test for each gene using the for loop but somehow, I also got the first gene in my list - "IRX1" assigned to a variable "gene". I think because of that, for my second loop, I am unable to get my boxplots because for R. I tried changing gene to a or i or something else, but the same thing kept happening. I will try and figure this out again as we go along. 

```

Comment:
I actually had a similar issue with encountering problems in the global environment.  I had to keep clearing the environment (closely R completely) and then re-run the code.  My issue was happening if I tried to run a chunk of code more than once I would get some weird things happening with variables and I'm not really sure why, not sure if this could also have been happening to you if you understand what I am saying.